### PR TITLE
Normalize and document handling of config options setting through env vars

### PR DESCRIPTION
**Pull request by remicalixte**
### What does this PR do?

Enable all configuration options to be set with env vars.

### Motivation

The goal is to make the configuration more consistent: All main options can be set either by the `datadog.yml` file or env vars. Env binding is done with default setting to avoid duplication.

### CODE REVIEW

File: pkg/config/config.go
Comment (hush-hush) [inline pkg/config/config.go]: Why empty string as a default and not `nil` ?

**Review by hush-hush — COMMENTED** [review]
Small question but this LGTM. Let's wait for @DataDog/container-integrations review before merging.

File: pkg/config/config.go
Comment (remicalixte) [reply pkg/config/config.go]: I chose "" because it's the zero of the string type

File: pkg/config/config.go
Comment (devonboyer) [inline pkg/config/config.go]: For `GetStringMapString` to work correctly these env vars need to be `json`. Should that be documented anywhere?

Comment (remicalixte) [reply pkg/config/config.go]: Oh yes, I'll add that

Comment (remicalixte) [reply pkg/config/config.go]: done :)

File: docs/agent/config.md
Comment (hush-hush) [inline docs/agent/config.md]: Are we sure ? I don't think you can set the proxy setting with a JSON

Comment (remicalixte) [reply docs/agent/config.md]: No it doesn't work for the proxy setting

Comment (hush-hush) [reply docs/agent/config.md]: So we shouldn't say `Any map structure should be json-formatted` then.

Comment (remicalixte) [reply docs/agent/config.md]: Yes actually it's any map structure but the proxy setting

Comment (remicalixte) [reply docs/agent/config.md]: I've updated the docs, hopefully it is clearer

File: docs/agent/config.md
Comment (hush-hush) [inline docs/agent/config.md]: People can use `DD_PROXY_HTTP`, `DD_PROXY_HTTPS` and `DD_PROXY_NO_PROXY` to set the proxy: https://github.com/DataDog/datadog-agent/blob/master/docs/agent/changes.md#proxies

@l0k0ms, @truthbk: could you give a look to the doc update ? Not sure what is the best way to word it.

File: docs/agent/config.md
Comment (l0k0ms) [inline docs/agent/config.md]: r / `Giving a nested variables by an environment variable will override all the others nested variables of the parameter specified in the configuration file.` / `Specifying nested variables with an environment variable overrides all the others nested variables of the parameter specified in the corresponding configuration file.`

File: docs/agent/config.md
Comment (l0k0ms) [reply docs/agent/config.md]: We should wait for the proxy doc to be released: https://github.com/DataDog/documentation/pull/2865 and link to it on the public doc.

Rephrasing would then be: 
r / ```The above note doesn't apply for the proxy setting. (We can specify `DD_PROXY_HTTP` through an environment variable and `proxy.no_proxy` in the configuration file but this is why it is also impossible to set the whole proxy setting through a json.)``` / 

`The above note doesn't apply for the proxy setting. [Refer to the dedicated Agent proxy documentation] to learn more.`

**Review by l0k0ms — COMMENTED** [review]
Added my comments

File: docs/agent/config.md
Comment (remicalixte) [reply docs/agent/config.md]: Okay, I'll add the above PR as a blocker

**Comment by remicalixte** [comment]
https://github.com/DataDog/documentation/pull/2865 was merged

**Review by hush-hush — DISMISSED** [review]

**Review by hush-hush — APPROVED** [review]

**Pull request merged**
